lib/integrations/CalDav/CalDavCalendarAdapter.ts

```diff
@@ -70,33 +68,26 @@ export class CalDavCalendar implements CalendarApiAdapter {
     return attendees.map(({ email, name }) => ({ name, email, partstat: "NEEDS-ACTION" }));
   }
 
-  async createEvent(event: CalendarEvent): Promise<Record<string, unknown>> {
+  async createEvent(event: CalendarEvent) {
     try {
       const calendars = await this.listCalendars();
       const uid = uuidv4();
 
-      const { error, value: iCalString } = await createEvent({
+      const { error, value: iCalString } = createEvent({
         uid,
         startInputType: "utc",
-        // FIXME types
         start: this.convertDate(event.startTime),
         duration: this.getDuration(event.startTime, event.endTime),
         title: event.title,
         description: event.description ?? "",
-        location: event.location,
+        location: event.location !== null ? event.location : undefined,
         organizer: { email: event.organizer.email, name: event.organizer.name },
         attendees: this.getAttendees(event.attendees),
       });
 
-      if (error) {
-        log.debug("Error creating iCalString");
-        return {};
-      }
+      if (error) throw new Error("Error creating iCalString");
 
-      if (!iCalString) {
-        log.debug("Error creating iCalString");
-        return {};
-      }
+      if (!iCalString) throw new Error("Error creating iCalString");
 
```

Comment:
Same as above. CalDav is kinda broken. This "may" fix it (or not). We should test.